--!strict
local Framework = require("../Framework")
local Md5 = require("../../src/Md5")

return function()
	Framework.describe("Md5.Hash", function()
		Framework.it("hashes empty string to d41d8cd98f00b204e9800998ecf8427e", function()
			Framework.expect(Md5.Hash("")).toBe("d41d8cd98f00b204e9800998ecf8427e")
		end)
		Framework.it("hashes 'a' correctly", function()
			Framework.expect(Md5.Hash("a")).toBe("0cc175b9c0f1b6a831c399e269772661")
		end)
		Framework.it("hashes 'abc' correctly", function()
			Framework.expect(Md5.Hash("abc")).toBe("900150983cd24fb0d6963f7d28e17f72")
		end)
		Framework.it("hashes 'message digest'", function()
			Framework.expect(Md5.Hash("message digest")).toBe("f96b697d7cb7938d525a2f31aaf161d0")
		end)
		Framework.it("hashes the lazy dog vector", function()
			Framework.expect(Md5.Hash("The quick brown fox jumps over the lazy dog")).toBe(
				"9e107d9d372bb6826bd81d3542a419d6"
			)
		end)
		Framework.it("output is always 32 hex chars lowercase", function()
			local h = Md5.Hash("anything")
			Framework.expect(#h).toBe(32)
			Framework.expect(string.match(h, "^[0-9a-f]+$") ~= nil).toBeTruthy()
		end)
	end)
end
